docs/advanced_topics/icons.md

```diff
@@ -0,0 +1,90 @@
+(icons)=
+
+# Icons
+
+Wagtail comes with an icon set. The icons are used throughout the admin interface.
+
+Elements that use icons are:
+
+- [Register Admin Menu Item](register_admin_menu_item)
+- [Clientside components](extending_clientside_components)
+- [Rich text editor toolbar buttons](extending_the_draftail_editor)
+- [ModelAdmin menu](modeladmin_menu_icon)
+- [Streamfield blocks](custom_streamfield_blocks)
+
+This document describes how to choose and add icons.
+
+## Available icons and their names
+
+Icons are registered in `wagtail/admin/wagtail_hooks.py`. 
+The _filename_ without the extension is the icon name.
+
+Enable the [styleguide](styleguide) to view icons and their names.
+
+[//]: # (The code is included to present an up-to-date icon name list)
+
+```{eval-rst}
+.. literalinclude:: ../../wagtail/admin/wagtail_hooks.py
+   :language: python
+   :pyobject: register_icons
+```
+
+## Add a custom icon
+
+Draw or download an icon.
```

Comment:
A huge "gotcha" is that the icon file will need to be put in a TEMPLATE folder, not a static folder. This took me several hours of head-scratching to figure out for the first time. I think it would be important to instruct the reader where to save the SVG file.

Reply:
There was some talk in https://github.com/wagtail/wagtail/issues/7511#issuecomment-1284975058 about moving the icon files out of templates and into static files, and from the sound of things it might not even be a huge task at this point... it would be really, really nice if we could make that change before this becomes a publicly documented API.

Reply:
Icons are stored as individual files (easier to maintain, override and extend) but are served as _a single sprite_. We don't need (or want?) to serve individual icon files.

I do get the confusion. But I fail to see the benefits of handling icons as static files. The staticfiles app just brings redundant features and complexity.

I've re-written the docs and explicitly mention that the SVG should go into a template folder.

Maybe I'm missing something. I'm happy to be convinced.